### PR TITLE
drive synchronizer

### PR DESCRIPTION
**Pull request by jenni**
DriveSynchronizer broken into methods / uploads now one backup at a time

### CODE REVIEW

File: lib/drive_synchronizer.rb [hunk 1]
```diff
@@ -3,22 +3,27 @@ class DriveSynchronizer < Synchronizer
...
-      session = GoogleDrive::Session.from_service_account_key(StringIO.new(@api_key))
+      session = session_meth
+      full_path = backup.path
+      filename = backup.filename
+      file = session.upload_from_file(full_path, filename)
+      add_to_folder(session, file)
+      session.root_collection.remove(file)
+    end
+
+    def session_meth
```
Comment (eviltrout) [inline lib/drive_synchronizer.rb]: You might want to just call this session?

Or you could assign it to an instance variable in the constructor `@session`

File: lib/synchronizer.rb [hunk 2]
```diff
@@ -2,25 +2,29 @@ class Synchronizer
...
-    @turned_on          = SiteSetting.discourse_backups_enabled
-    @number_of_backups  = SiteSetting.discourse_backups_quantity
-	end
+  def backup
+    @backup
+  end
+
+  def initialize(backup)
+    @backup = backup
+    @api_key = SiteSetting.discourse_backups_api_key
```
Comment (eviltrout) [inline lib/synchronizer.rb:11]: I think the issue here is that `can_sync?` should not be implemented in the base class. It should only be implemented by the class that extends it. So you don't need to check site settings in this class, let the classes that extend this (GoogleDriveSynchronizer) do all that, using its own site settings.

File: lib/drive_synchronizer.rb (hunk 1 above)
Comment (jen-v) [inline lib/drive_synchronizer.rb]: https://github.com/berlindiamonds/discourse-googledrive-backup/blob/31391e9d32afe23f29121be737b60d1b2d9175d7/lib/drive_synchronizer.rb

just added @session to the constructor

Comment (jenni) [reply lib/drive_synchronizer.rb]: https://github.com/berlindiamonds/discourse-googledrive-backup/pull/23/commits/ec95575e2d3b3f74aa325e3586fb55eadd2314ef

got excited discovering the "super" keyword, feels like **magic**!

**Pull request merged**